lib/views/setting/setting_view.dart

```diff
@@ -37,77 +39,65 @@ class _SettingViewState extends State<SettingView> {
 
   Future<String> pickDir() async {
     String? path = await FilePicker.platform.getDirectoryPath();
+    // FIXME: migrate the setting to null safety, use the null to avoid
+    // check on string
     path == null ? setting.path = "No path found" : setting.path = path;
     return setting.path;
   }
 
   Future<bool> saveSettings() async {
     final prefs = await SharedPreferences.getInstance();
-
-    clients.asMap().forEach((index, element) async {
-      if (setting.connectionType == element) {
-        await prefs.setInt('connectionClient', index);
-      }
-    });
-
-    await prefs.setString('selectedPath', setting.path);
-
-    await prefs.setString('host', setting.host);
-
-    await prefs.setString(
-        'nickName', setting.nickName.isEmpty ? "null" : setting.nickName);
-
+    await prefs.setString("setting", json.encode(setting.toJson()));
     return true;
   }
 
-  Widget _buildMainView({required BuildContext context}) {
-    return Padding(
-      padding: EdgeInsets.only(
-          left: MediaQuery.of(context).size.width * 0.2,
-          right: MediaQuery.of(context).size.width * 0.2),
-      child: Column(
-        crossAxisAlignment: CrossAxisAlignment.start,
-        mainAxisAlignment: MainAxisAlignment.spaceAround,
-        children: [
-          const Text("Connection type"),
+  Widget _buildGrpcSettingView({required BuildContext context}) {
+    return Wrap(
+        runSpacing: MediaQuery.of(context).size.height * 0.05,
+        children: <Widget>[
```

Comment:
I was thinking that we can move this to another widget class? each setting view can be a new widget? What do you think? The change should be also easy